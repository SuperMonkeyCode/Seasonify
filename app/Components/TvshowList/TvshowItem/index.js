/* @flow */
import { connect } from 'react-redux';

import { tvshowSelectors } from 'Store/tvshowStore';
import { viewingSelectors } from 'Store/viewingStore';

import TvshowItem from './TvshowItem';

export default connect(
  () => (state, { tvshowId }) => ({
    seasonsCount: tvshowSelectors.makeGetSeasonsCount()(state, { tvshowId }),
    tvshow: tvshowSelectors.getTvshow(state, { tvshowId }),
    viewers: viewingSelectors.makeGetViewersArray()(state, { tvshowId }),
  }),
  {},
)(TvshowItem);
