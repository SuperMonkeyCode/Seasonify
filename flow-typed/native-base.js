/* @flow */
declare module 'native-base' {
  declare var exports: any;
}
