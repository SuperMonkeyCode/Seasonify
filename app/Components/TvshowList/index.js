/* @flow */
import { connect } from 'react-redux';
import React from 'react';

import HeaderRoot from 'Components/HeaderRoot';
import tvshowState from 'State/tvshowState';

import TvshowList from './TvshowList';

TvshowList.navigationOptions = ({ navigation }) => ({
  header: <HeaderRoot title="TV Shows" navigation={navigation} />,
});

export default connect(
  state => ({
    tvshows: tvshowState.selectors.getTvshows(state),
  }),
  {},
)(TvshowList);
