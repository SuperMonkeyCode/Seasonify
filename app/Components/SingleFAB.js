/* @flow */
import { Fab, Icon } from 'native-base';
import React from 'react';

export default function SingleFAB({ icon, onPress }: { icon: string, onPress: Function }) {
  return (
    <Fab
      // containerStyle={{ marginRight: 10 }}
      style={{ backgroundColor: '#5067FF' }}
      position="bottomRight"
      onPress={onPress}
    >
      <Icon name={icon} />
    </Fab>
  );
}
