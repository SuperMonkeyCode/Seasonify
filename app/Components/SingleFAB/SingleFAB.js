/* @flow */
import { Fab, Icon } from 'native-base';
import { pure } from 'recompose';
import React from 'react';

type Props = {
  /* parent */
  icon: string,
  onPress: Function,
  /* connect */
  /* HOC */
};

const enhance = pure;

function SingleFAB({ icon, onPress }: Props) {
  return (
    <Fab
      // containerStyle={{ marginRight: 10 }}
      style={{ backgroundColor: '#5067FF' }}
      position="bottomRight"
      onPress={onPress}
    >
      <Icon name={icon} />
    </Fab>
  );
}

export default enhance(SingleFAB);
