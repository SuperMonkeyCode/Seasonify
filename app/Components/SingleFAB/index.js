/* @flow */
import SingleFAB from './SingleFAB';

export default SingleFAB;
