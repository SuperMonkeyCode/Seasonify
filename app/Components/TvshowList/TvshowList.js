/* @flow */
import { Container, Content } from 'native-base';
import { /* FlatList,*/ VirtualizedList } from 'react-native';
import {
  branch,
  compose,
  hoistStatics,
  pure,
  renderComponent,
  withHandlers,
} from 'recompose';
import React from 'react';

import Loading from 'Components/Loading';
import SingleFAB from 'Components/SingleFAB';
import TvshowItem from './TvshowItem';

type Props = {
  /* parent */
  navigation: Object,
  /* connect */
  tvshowsIds: string[],
  isLoading: LoadingValue,
  /* HOC */
  handleFAB: Function,
};

const enhance = compose(
  pure,
  withHandlers({
    handleFAB: ({ navigation }: Props) => () => {
      navigation.navigate('TvshowAddPage');
    },
  }),
  branch(({ isLoading }: Props) => isLoading, renderComponent(Loading)),
);

function TvshowList({ navigation, tvshowsIds, handleFAB }: Props) {
  /* istanbul ignore next */
  const renderItem = ({ item: id }) =>
    <TvshowItem tvshowId={id} navigate={navigation.navigate} />;

  const getItem = (data, index) => data[index];
  const keyExtractor = item => item;

  return (
    <Container>
      <Content>
        <VirtualizedList
          initialNumberToRender={11}
          data={tvshowsIds}
          getItemCount={/* istanbul ignore next */ data => data.length}
          getItem={getItem}
          renderItem={renderItem}
          keyExtractor={keyExtractor}
        />
      </Content>
      <SingleFAB icon="add" onPress={handleFAB} />
    </Container>
  );
}

export default enhance(TvshowList);
