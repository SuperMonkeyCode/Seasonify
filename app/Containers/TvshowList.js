/* @flow */
import { Container, Content } from 'native-base';
import { /* FlatList,*/ VirtualizedList } from 'react-native';
import { connect } from 'react-redux';
import React from 'react';

import type { Tvshows } from 'Types';
import HeaderRoot from 'Components/HeaderRoot';
import SingleFAB from 'Components/SingleFAB';
import TvshowItem from 'Components/TvshowItem';
import tvshowState from 'State/tvshowState';

const mapStateToProps = state => ({
  tvshows: tvshowState.selectors.getTvshows(state),
});

const mapActionsToProps = {};

export function TvshowList({ navigation, tvshows }: { navigation: Object, tvshows: Tvshows }) {
  const onFAB = () => {
    navigation.navigate('TvshowAddPage');
  };

  /* istanbul ignore next */
  const renderItem = ({ item: tvshow }) =>
    (<TvshowItem
      tvshowId={tvshow.id}
      tvshowName={tvshow.name}
      navigate={navigation.navigate}
    />);

  const keyExtractor = item => item.id;

  const tvshowKeys = Object.keys(tvshows);

  const getItem = (data, index) => data[tvshowKeys[index]];

  return (
    <Container>
      <Content>
        <VirtualizedList
          initialNumberToRender={11}
          data={tvshows}
          extraData={tvshowKeys}
          getItemCount={/* istanbul ignore next */ data => Object.keys(data).length}
          getItem={getItem}
          renderItem={renderItem}
          keyExtractor={keyExtractor}
        />
      </Content>
      <SingleFAB icon="add" onPress={onFAB} />
    </Container>
  );
}

TvshowList.navigationOptions = ({ navigation }) => ({
  header: <HeaderRoot title="TV Shows" navigation={navigation} />,
});

export default connect(mapStateToProps, mapActionsToProps)(TvshowList);
