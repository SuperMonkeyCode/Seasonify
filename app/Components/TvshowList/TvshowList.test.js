import { Container } from 'native-base';
import React from 'react';

import SingleFAB from 'Components/SingleFAB';

import TvshowList from './TvshowList';

function setup(specificProps = {}) {
  const props = {
    tvshows: {},
    navigation: { navigate: jest.fn() },
    ...specificProps,
  };
  const component = shallowDive(<TvshowList {...props} />, Container);
  return {
    component,
    props,
    actions: {},
  };
}

describe('rendering', () => {
  it('should show 2 series', () => {
    const { component } = setup({
      tvshows: {
        abc: { id: 'abc', name: 'tvshow1' },
        qsd: { id: 'qsd', name: 'tvshow2' },
      },
    });
    expect(component).toMatchSnapshot();
  });

  describe('when clicking on FAB', () => {
    it('should navigate', () => {
      const { component, props } = setup();
      component.find(SingleFAB).simulate('press');
      expect(props.navigation.navigate).toBeCalledWith('TvshowAddPage');
    });
  });
});
