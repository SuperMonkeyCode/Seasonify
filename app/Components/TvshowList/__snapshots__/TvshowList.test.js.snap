// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering should show 2 series 1`] = `
<Styled(Container)>
  <Styled(Content)>
    <VirtualizedList
      data={
        Object {
          "abc": Object {
            "id": "abc",
            "name": "tvshow1",
          },
          "qsd": Object {
            "id": "qsd",
            "name": "tvshow2",
          },
        }
      }
      disableVirtualization={false}
      extraData={
        Array [
          "abc",
          "qsd",
        ]
      }
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      initialNumberToRender={11}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      onEndReached={[Function]}
      onEndReachedThreshold={2}
      removeClippedSubviews={true}
      renderItem={[Function]}
      renderScrollComponent={[Function]}
      updateCellsBatchingPeriod={50}
      windowSize={21}
    />
  </Styled(Content)>
  <pure(SingleFAB)
    icon="add"
    onPress={[Function]}
  />
</Styled(Container)>
`;
